<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!--NewPage-->
<HTML>
<HEAD>
<!-- Generated by javadoc (build 1.6.0_21) on Thu Sep 05 14:48:23 EDT 2013 -->
<TITLE>
Uses of Class org.helm.notation.NotationException (HELMNotationToolkit v1.0.1)
</TITLE>

<META NAME="date" CONTENT="2013-09-05">

<LINK REL ="stylesheet" TYPE="text/css" HREF="../../../../stylesheet.css" TITLE="Style">

<SCRIPT type="text/javascript">
function windowTitle()
{
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="Uses of Class org.helm.notation.NotationException (HELMNotationToolkit v1.0.1)";
    }
}
</SCRIPT>
<NOSCRIPT>
</NOSCRIPT>

</HEAD>

<BODY BGCOLOR="white" onload="windowTitle();">
<HR>


<!-- ========= START OF TOP NAVBAR ======= -->
<A NAME="navbar_top"><!-- --></A>
<A HREF="#skip-navbar_top" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_top_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../package-summary.html"><FONT CLASS="NavBarFont1"><B>Package</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../org/helm/notation/NotationException.html" title="class in org.helm.notation"><FONT CLASS="NavBarFont1"><B>Class</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Use</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../package-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../index-files/index-1.html"><FONT CLASS="NavBarFont1"><B>Index</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
<b>HELMNotationToolkit<br><font size='-1'>1.0.1</font></b></EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;PREV&nbsp;
&nbsp;NEXT</FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../../../../index.html?org/helm/notation/\class-useNotationException.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="NotationException.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../../../../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../../../../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_top"></A>
<!-- ========= END OF TOP NAVBAR ========= -->

<HR>
<CENTER>
<H2>
<B>Uses of Class<br>org.helm.notation.NotationException</B></H2>
</CENTER>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Packages that use <A HREF="../../../../org/helm/notation/NotationException.html" title="class in org.helm.notation">NotationException</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#org.helm.notation"><B>org.helm.notation</B></A></TD>
<TD>&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#org.helm.notation.tools"><B>org.helm.notation.tools</B></A></TD>
<TD>&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="org.helm.notation"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Uses of <A HREF="../../../../org/helm/notation/NotationException.html" title="class in org.helm.notation">NotationException</A> in <A HREF="../../../../org/helm/notation/package-summary.html">org.helm.notation</A></FONT></TH>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Methods in <A HREF="../../../../org/helm/notation/package-summary.html">org.helm.notation</A> that throw <A HREF="../../../../org/helm/notation/NotationException.html" title="class in org.helm.notation">NotationException</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;<A HREF="../../../../org/helm/notation/NucleotideFactory.html" title="class in org.helm.notation">NucleotideFactory</A></CODE></FONT></TD>
<TD><CODE><B>NucleotideFactory.</B><B><A HREF="../../../../org/helm/notation/NucleotideFactory.html#getInstance()">getInstance</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="org.helm.notation.tools"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Uses of <A HREF="../../../../org/helm/notation/NotationException.html" title="class in org.helm.notation">NotationException</A> in <A HREF="../../../../org/helm/notation/tools/package-summary.html">org.helm.notation.tools</A></FONT></TH>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Methods in <A HREF="../../../../org/helm/notation/tools/package-summary.html">org.helm.notation.tools</A> that throw <A HREF="../../../../org/helm/notation/NotationException.html" title="class in org.helm.notation">NotationException</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;float</CODE></FONT></TD>
<TD><CODE><B>ExtinctionCoefficientCalculator.</B><B><A HREF="../../../../org/helm/notation/tools/ExtinctionCoefficientCalculator.html#calculateFromComplexNotation(java.lang.String)">calculateFromComplexNotation</A></B>(java.lang.String&nbsp;complexNotation)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;This method calculates extinction coefficient for complex polymer notation</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;float</CODE></FONT></TD>
<TD><CODE><B>ExtinctionCoefficientCalculator.</B><B><A HREF="../../../../org/helm/notation/tools/ExtinctionCoefficientCalculator.html#calculateFromComplexNotation(java.lang.String, int)">calculateFromComplexNotation</A></B>(java.lang.String&nbsp;complexNotation,
                             int&nbsp;unitType)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;This method calculates extinction coefficient for complex polymer notation</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;float</CODE></FONT></TD>
<TD><CODE><B>ExtinctionCoefficientCalculator.</B><B><A HREF="../../../../org/helm/notation/tools/ExtinctionCoefficientCalculator.html#calculateFromModifiedNucleotideSequence(java.lang.String)">calculateFromModifiedNucleotideSequence</A></B>(java.lang.String&nbsp;sequence)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;This method calculates extinction coefficient for modified nucleotide sequence</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;float</CODE></FONT></TD>
<TD><CODE><B>ExtinctionCoefficientCalculator.</B><B><A HREF="../../../../org/helm/notation/tools/ExtinctionCoefficientCalculator.html#calculateFromPeptidePolymerNotation(java.lang.String)">calculateFromPeptidePolymerNotation</A></B>(java.lang.String&nbsp;simpleNotation)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;float</CODE></FONT></TD>
<TD><CODE><B>ExtinctionCoefficientCalculator.</B><B><A HREF="../../../../org/helm/notation/tools/ExtinctionCoefficientCalculator.html#calculateFromRnaPolymerNotation(java.lang.String)">calculateFromRnaPolymerNotation</A></B>(java.lang.String&nbsp;simpleNotation)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;This method calculates extinction coefficient for RNA polymer notation</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;void</CODE></FONT></TD>
<TD><CODE><B>EdgeParser.</B><B><A HREF="../../../../org/helm/notation/tools/EdgeParser.html#canonicalize(org.helm.notation.model.PolymerEdge, java.util.List)">canonicalize</A></B>(<A HREF="../../../../org/helm/notation/model/PolymerEdge.html" title="class in org.helm.notation.model">PolymerEdge</A>&nbsp;polymerEdge,
             java.util.List&lt;java.lang.String&gt;&nbsp;sortedNodeIDs)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;canonicalize input polymerEdge based on sortedNodeIDs</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B>ComplexNotationParser.</B><B><A HREF="../../../../org/helm/notation/tools/ComplexNotationParser.html#containsGenericStructure(java.lang.String)">containsGenericStructure</A></B>(java.lang.String&nbsp;extendendNotation)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;This method determines if the extendendNotation contains generic structure (cannot generate SMILES)</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;<A HREF="../../../../org/helm/notation/model/Nucleotide.html" title="class in org.helm.notation.model">Nucleotide</A></CODE></FONT></TD>
<TD><CODE><B>DeepCopy.</B><B><A HREF="../../../../org/helm/notation/tools/DeepCopy.html#copy(org.helm.notation.model.Nucleotide)">copy</A></B>(<A HREF="../../../../org/helm/notation/model/Nucleotide.html" title="class in org.helm.notation.model">Nucleotide</A>&nbsp;nucleotide)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.io.Serializable</CODE></FONT></TD>
<TD><CODE><B>DeepCopy.</B><B><A HREF="../../../../org/helm/notation/tools/DeepCopy.html#copy(java.io.Serializable)">copy</A></B>(java.io.Serializable&nbsp;input)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.lang.String[]</CODE></FONT></TD>
<TD><CODE><B>ComplexNotationParser.</B><B><A HREF="../../../../org/helm/notation/tools/ComplexNotationParser.html#decompose(java.lang.String)">decompose</A></B>(java.lang.String&nbsp;complexNotation)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;This method decomposes complex notation into complex notations of covalently connected monomers, base pairing info is lost</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B>ComplexNotationParser.</B><B><A HREF="../../../../org/helm/notation/tools/ComplexNotationParser.html#getAllBasePairString(java.lang.String)">getAllBasePairString</A></B>(java.lang.String&nbsp;extendedNotation)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B>ComplexNotationParser.</B><B><A HREF="../../../../org/helm/notation/tools/ComplexNotationParser.html#getAllEdgeString(java.lang.String)">getAllEdgeString</A></B>(java.lang.String&nbsp;extendedNotation)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B>ComplexNotationParser.</B><B><A HREF="../../../../org/helm/notation/tools/ComplexNotationParser.html#getAllNodeLabelString(java.lang.String)">getAllNodeLabelString</A></B>(java.lang.String&nbsp;extendedNotation)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B>ComplexNotationParser.</B><B><A HREF="../../../../org/helm/notation/tools/ComplexNotationParser.html#getAllNodeString(java.lang.String)">getAllNodeString</A></B>(java.lang.String&nbsp;extendedNotation)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.util.List&lt;java.lang.String&gt;</CODE></FONT></TD>
<TD><CODE><B>PeptideSequenceParser.</B><B><A HREF="../../../../org/helm/notation/tools/PeptideSequenceParser.html#getAminoAcidList(java.lang.String)">getAminoAcidList</A></B>(java.lang.String&nbsp;peptideSequence)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;This method converts peptide sequence into a List of amino acid</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.util.List&lt;java.lang.String&gt;</CODE></FONT></TD>
<TD><CODE><B>PeptideSequenceParser.</B><B><A HREF="../../../../org/helm/notation/tools/PeptideSequenceParser.html#getAminoAcidList(java.lang.String, java.lang.String)">getAminoAcidList</A></B>(java.lang.String&nbsp;peptideSequence,
                 java.lang.String&nbsp;delimiter)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;This method converts peptide sequence into a List of amino acid with optional delimiter</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B>ComplexNotationParser.</B><B><A HREF="../../../../org/helm/notation/tools/ComplexNotationParser.html#getCanonicalNotation(java.lang.String)">getCanonicalNotation</A></B>(java.lang.String&nbsp;complexNotation)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Generate canonical notation without validation</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B>ComplexNotationParser.</B><B><A HREF="../../../../org/helm/notation/tools/ComplexNotationParser.html#getCanonicalNotation(java.lang.String, boolean)">getCanonicalNotation</A></B>(java.lang.String&nbsp;complexNotation,
                     boolean&nbsp;includeValidation)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Generate canonical notation for the the input complex notation</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B>ComplexNotationParser.</B><B><A HREF="../../../../org/helm/notation/tools/ComplexNotationParser.html#getCombinedComlexNotation(java.lang.String, java.lang.String)">getCombinedComlexNotation</A></B>(java.lang.String&nbsp;complexNotation1,
                          java.lang.String&nbsp;complexNotation2)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;The method generates the complex the notation for two complex notations combined.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B>NucleotideConverter.</B><B><A HREF="../../../../org/helm/notation/tools/NucleotideConverter.html#getComplexNotation(java.lang.String)">getComplexNotation</A></B>(java.lang.String&nbsp;nucleotideSequences)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;This method return the HELM complex notation for nucleotide sequences, which could have more than one strand separated by space</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B>SimpleNotationParser.</B><B><A HREF="../../../../org/helm/notation/tools/SimpleNotationParser.html#getComplexNotation(java.lang.String, java.lang.String)">getComplexNotation</A></B>(java.lang.String&nbsp;simpleNotation,
                   java.lang.String&nbsp;polymerType)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;This method converts simple notation for a given polymer type to complex notation.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B>NucleotideConverter.</B><B><A HREF="../../../../org/helm/notation/tools/NucleotideConverter.html#getComplexNotationFromSingleNucleotideSequence(java.lang.String)">getComplexNotationFromSingleNucleotideSequence</A></B>(java.lang.String&nbsp;singleOligoSequence)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;This method returns the complex notation for a single oligonucleotide sequence</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B>ComplexNotationParser.</B><B><A HREF="../../../../org/helm/notation/tools/ComplexNotationParser.html#getComplexPolymerCanonicalSmiles(java.lang.String)">getComplexPolymerCanonicalSmiles</A></B>(java.lang.String&nbsp;extendedNotation)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B>ComplexNotationParser.</B><B><A HREF="../../../../org/helm/notation/tools/ComplexNotationParser.html#getComplexPolymerSMILES(java.lang.String)">getComplexPolymerSMILES</A></B>(java.lang.String&nbsp;extendedNotation)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;This methods returns the unique SMILES string for complex polymer extendendNotation, if all monomers have specific structures</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.util.List&lt;chemaxon.struc.Molecule&gt;</CODE></FONT></TD>
<TD><CODE><B>ComplexNotationParser.</B><B><A HREF="../../../../org/helm/notation/tools/ComplexNotationParser.html#getComplexPolymerStructure(java.lang.String)">getComplexPolymerStructure</A></B>(java.lang.String&nbsp;extendedNotation)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;This method returns list of Molecule with specific structure, all R groups are filled</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B>SimpleNotationParser.</B><B><A HREF="../../../../org/helm/notation/tools/SimpleNotationParser.html#getComplextNotationForAntisenseRNA(java.lang.String)">getComplextNotationForAntisenseRNA</A></B>(java.lang.String&nbsp;simpleNotation)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;convert RNA simple notation to complex notation and annotate as antisense strand 'as'</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B>SimpleNotationParser.</B><B><A HREF="../../../../org/helm/notation/tools/SimpleNotationParser.html#getComplextNotationForChem(java.lang.String)">getComplextNotationForChem</A></B>(java.lang.String&nbsp;simpleNotation)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;convert chemical simple notation to complex notation</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B>SimpleNotationParser.</B><B><A HREF="../../../../org/helm/notation/tools/SimpleNotationParser.html#getComplextNotationForPeptide(java.lang.String)">getComplextNotationForPeptide</A></B>(java.lang.String&nbsp;simpleNotation)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;convert Peptide simple notation to complex notation</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B>SimpleNotationParser.</B><B><A HREF="../../../../org/helm/notation/tools/SimpleNotationParser.html#getComplextNotationForRNA(java.lang.String)">getComplextNotationForRNA</A></B>(java.lang.String&nbsp;simpleNotation)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;convert RNA simple notation to complex notation</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B>SimpleNotationParser.</B><B><A HREF="../../../../org/helm/notation/tools/SimpleNotationParser.html#getComplextNotationForSenseRNA(java.lang.String)">getComplextNotationForSenseRNA</A></B>(java.lang.String&nbsp;simpleNotation)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;convert RNA simple notation to complex notation and annotate as sense strand 'ss'</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.lang.String[]</CODE></FONT></TD>
<TD><CODE><B>ComplexNotationParser.</B><B><A HREF="../../../../org/helm/notation/tools/ComplexNotationParser.html#getFormatedSirnaSequences(java.lang.String)">getFormatedSirnaSequences</A></B>(java.lang.String&nbsp;complexNotaton)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;generate formated siRNA sequence with default padding char " " and base-pair char "|"</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.lang.String[]</CODE></FONT></TD>
<TD><CODE><B>ComplexNotationParser.</B><B><A HREF="../../../../org/helm/notation/tools/ComplexNotationParser.html#getFormatedSirnaSequences(java.lang.String, java.lang.String, java.lang.String)">getFormatedSirnaSequences</A></B>(java.lang.String&nbsp;complexNotation,
                          java.lang.String&nbsp;paddingChar,
                          java.lang.String&nbsp;basePairChar)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;generates formated siRNA sequence with provided padding and base-pair character, such as
   AGCUUUGGTT
   ||||||||
 TTUCGAAACC</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;<A HREF="../../../../org/helm/notation/tools/NucleotideConverter.html" title="class in org.helm.notation.tools">NucleotideConverter</A></CODE></FONT></TD>
<TD><CODE><B>NucleotideConverter.</B><B><A HREF="../../../../org/helm/notation/tools/NucleotideConverter.html#getInstance()">getInstance</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B>NucleotideSequenceParser.</B><B><A HREF="../../../../org/helm/notation/tools/NucleotideSequenceParser.html#getMaxMatchFragment(java.lang.String, java.lang.String)">getMaxMatchFragment</A></B>(java.lang.String&nbsp;seq1,
                    java.lang.String&nbsp;seq2)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B>NucleotideSequenceParser.</B><B><A HREF="../../../../org/helm/notation/tools/NucleotideSequenceParser.html#getMaxMatchFragment(java.lang.String, java.lang.String, int)">getMaxMatchFragment</A></B>(java.lang.String&nbsp;seq1,
                    java.lang.String&nbsp;seq2,
                    int&nbsp;minLength)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;This method returns the largest matched fragment between two sequences, replace T with U before match</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>protected static&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B>SimpleNotationParser.</B><B><A HREF="../../../../org/helm/notation/tools/SimpleNotationParser.html#getModifiedNucleotideSequence(java.util.List)">getModifiedNucleotideSequence</A></B>(java.util.List&lt;<A HREF="../../../../org/helm/notation/model/Nucleotide.html" title="class in org.helm.notation.model">Nucleotide</A>&gt;&nbsp;nucleotideList)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B>SimpleNotationParser.</B><B><A HREF="../../../../org/helm/notation/tools/SimpleNotationParser.html#getModifiedNucleotideSequence(java.lang.String)">getModifiedNucleotideSequence</A></B>(java.lang.String&nbsp;polymerNotation)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;convert RNA simple notation to modified nucleotide sequence</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B>SimpleNotationParser.</B><B><A HREF="../../../../org/helm/notation/tools/SimpleNotationParser.html#getModifiedPeptideSequence(java.lang.String, java.lang.String)">getModifiedPeptideSequence</A></B>(java.lang.String&nbsp;polymerNotation,
                           java.lang.String&nbsp;delimiter)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;This method returns the modified peptide sequence</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;<A HREF="../../../../org/helm/notation/model/MoleculeInfo.html" title="class in org.helm.notation.model">MoleculeInfo</A></CODE></FONT></TD>
<TD><CODE><B>ComplexNotationParser.</B><B><A HREF="../../../../org/helm/notation/tools/ComplexNotationParser.html#getMoleculeInfo(java.lang.String)">getMoleculeInfo</A></B>(java.lang.String&nbsp;extendedNotation)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Perform calculation without validation</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;<A HREF="../../../../org/helm/notation/model/MoleculeInfo.html" title="class in org.helm.notation.model">MoleculeInfo</A></CODE></FONT></TD>
<TD><CODE><B>ComplexNotationParser.</B><B><A HREF="../../../../org/helm/notation/tools/ComplexNotationParser.html#getMoleculeInfo(java.lang.String, boolean)">getMoleculeInfo</A></B>(java.lang.String&nbsp;extendedNotation,
                boolean&nbsp;includeValidation)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;This method returns the MoleculeInfo of input polymer notation using the divide and conquer approach in SimpleNotationParser
 Should return the same result as StructureParser.getMoleculeInfo(String smiles) method, but runs faster for standard edges
 Ignore fuzzy edges in the calculation, good enough for large structures</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;<A HREF="../../../../org/helm/notation/model/MoleculeInfo.html" title="class in org.helm.notation.model">MoleculeInfo</A></CODE></FONT></TD>
<TD><CODE><B>SimpleNotationParser.</B><B><A HREF="../../../../org/helm/notation/tools/SimpleNotationParser.html#getMoleculeInfo(java.lang.String, java.lang.String)">getMoleculeInfo</A></B>(java.lang.String&nbsp;notation,
                java.lang.String&nbsp;polymerType)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;This method returns the MoleculeInfo of simple polymer using a divide and conquer approach</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;<A HREF="../../../../org/helm/notation/model/Monomer.html" title="class in org.helm.notation.model">Monomer</A></CODE></FONT></TD>
<TD><CODE><B>SimpleNotationParser.</B><B><A HREF="../../../../org/helm/notation/tools/SimpleNotationParser.html#getMonomer(java.lang.String, java.lang.String)">getMonomer</A></B>(java.lang.String&nbsp;monomerID,
           java.lang.String&nbsp;polymerType)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;This method returns Monomer object based on monomer ID and polymer type</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;int</CODE></FONT></TD>
<TD><CODE><B>SimpleNotationParser.</B><B><A HREF="../../../../org/helm/notation/tools/SimpleNotationParser.html#getMonomerCount(java.lang.String, java.lang.String)">getMonomerCount</A></B>(java.lang.String&nbsp;simpleNotation,
                java.lang.String&nbsp;polymerType)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;This method returns the total number of monomers in a simple notation for a give polymer type</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;int</CODE></FONT></TD>
<TD><CODE><B>SimpleNotationParser.</B><B><A HREF="../../../../org/helm/notation/tools/SimpleNotationParser.html#getMonomerCountForPeptide(java.lang.String)">getMonomerCountForPeptide</A></B>(java.lang.String&nbsp;simpleNotation)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;This method returns the total number of monomers in peptide simple notation</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;int</CODE></FONT></TD>
<TD><CODE><B>SimpleNotationParser.</B><B><A HREF="../../../../org/helm/notation/tools/SimpleNotationParser.html#getMonomerCountForRNA(java.lang.String)">getMonomerCountForRNA</A></B>(java.lang.String&nbsp;simpleNotation)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;This method returns the total number of monomers in RNA simple notation</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.util.List&lt;java.lang.String&gt;</CODE></FONT></TD>
<TD><CODE><B>SimpleNotationParser.</B><B><A HREF="../../../../org/helm/notation/tools/SimpleNotationParser.html#getMonomerIDList(java.lang.String, java.lang.String)">getMonomerIDList</A></B>(java.lang.String&nbsp;polymerNotation,
                 java.lang.String&nbsp;polymerType)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;This method returns the list of Monomer IDs from simple polymer notation</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.util.List&lt;<A HREF="../../../../org/helm/notation/model/Monomer.html" title="class in org.helm.notation.model">Monomer</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>SimpleNotationParser.</B><B><A HREF="../../../../org/helm/notation/tools/SimpleNotationParser.html#getMonomerList(java.lang.String, java.lang.String)">getMonomerList</A></B>(java.lang.String&nbsp;polymerNotation,
               java.lang.String&nbsp;polymerType)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;This methods generates the list of Monomer from polymer notation</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B>NucleotideSequenceParser.</B><B><A HREF="../../../../org/helm/notation/tools/NucleotideSequenceParser.html#getNaturalAnalogSequence(java.lang.String)">getNaturalAnalogSequence</A></B>(java.lang.String&nbsp;sequence)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;convert modified sequence to natural analog sequence</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B>NucleotideSequenceParser.</B><B><A HREF="../../../../org/helm/notation/tools/NucleotideSequenceParser.html#getNormalComplementSequence(java.lang.String)">getNormalComplementSequence</A></B>(java.lang.String&nbsp;sequence)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;returns complement sequence in normal (5-3) direction</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B>NucleotideSequenceParser.</B><B><A HREF="../../../../org/helm/notation/tools/NucleotideSequenceParser.html#getNormalComplementSequence(java.lang.String, java.lang.String)">getNormalComplementSequence</A></B>(java.lang.String&nbsp;sequence,
                            java.lang.String&nbsp;notationSource)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.util.List&lt;<A HREF="../../../../org/helm/notation/model/Nucleotide.html" title="class in org.helm.notation.model">Nucleotide</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>NucleotideSequenceParser.</B><B><A HREF="../../../../org/helm/notation/tools/NucleotideSequenceParser.html#getNormalList(java.lang.String)">getNormalList</A></B>(java.lang.String&nbsp;sequence)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.util.List&lt;<A HREF="../../../../org/helm/notation/model/Nucleotide.html" title="class in org.helm.notation.model">Nucleotide</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>NucleotideSequenceParser.</B><B><A HREF="../../../../org/helm/notation/tools/NucleotideSequenceParser.html#getNormalList(java.lang.String, java.lang.String)">getNormalList</A></B>(java.lang.String&nbsp;sequence,
              java.lang.String&nbsp;notationSource)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;get the noraml nucleotide list regardness the direction of input sequence</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B>PeptideSequenceParser.</B><B><A HREF="../../../../org/helm/notation/tools/PeptideSequenceParser.html#getNotation(java.lang.String)">getNotation</A></B>(java.lang.String&nbsp;sequence)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;This methods converts peptide sequence into simple notation for peptide polymer</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B>NucleotideSequenceParser.</B><B><A HREF="../../../../org/helm/notation/tools/NucleotideSequenceParser.html#getNotation(java.lang.String)">getNotation</A></B>(java.lang.String&nbsp;sequence)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Assume sequence input in one of the following format<br>
 5'-AGCUdT-3'<br>
  3'-dTUCGA-5'<br>
   AFCUdT  (default 5' to 3')<br></TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B>PeptideSequenceParser.</B><B><A HREF="../../../../org/helm/notation/tools/PeptideSequenceParser.html#getNotation(java.lang.String, java.lang.String)">getNotation</A></B>(java.lang.String&nbsp;sequence,
            java.lang.String&nbsp;delimiter)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;This methods converts peptide sequence into simple notation for peptide polymer, with optional delimiter</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B>NucleotideSequenceParser.</B><B><A HREF="../../../../org/helm/notation/tools/NucleotideSequenceParser.html#getNotation(java.lang.String, java.lang.String)">getNotation</A></B>(java.lang.String&nbsp;sequence,
            java.lang.String&nbsp;notationSource)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.util.List&lt;<A HREF="../../../../org/helm/notation/model/Nucleotide.html" title="class in org.helm.notation.model">Nucleotide</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>SimpleNotationParser.</B><B><A HREF="../../../../org/helm/notation/tools/SimpleNotationParser.html#getNucleotideList(java.lang.String)">getNucleotideList</A></B>(java.lang.String&nbsp;polymerNotation)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;This methods returns the nucleotide list for RNA polymer type, with structure validation on.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.util.List&lt;<A HREF="../../../../org/helm/notation/model/Nucleotide.html" title="class in org.helm.notation.model">Nucleotide</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>SimpleNotationParser.</B><B><A HREF="../../../../org/helm/notation/tools/SimpleNotationParser.html#getNucleotideList(java.lang.String, boolean)">getNucleotideList</A></B>(java.lang.String&nbsp;polymerNotation,
                  boolean&nbsp;validate)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;This methods returns the nucleotide list for RNA polymer type.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B>SimpleNotationParser.</B><B><A HREF="../../../../org/helm/notation/tools/SimpleNotationParser.html#getNucleotideSequence(java.lang.String)">getNucleotideSequence</A></B>(java.lang.String&nbsp;polymerNotation)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;generate single letter (natural analog) nucleotide sequence for simple RNA notatoin</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B>NucleotideConverter.</B><B><A HREF="../../../../org/helm/notation/tools/NucleotideConverter.html#getNucleotideSequenceFromSimpleRNANotation(java.lang.String)">getNucleotideSequenceFromSimpleRNANotation</A></B>(java.lang.String&nbsp;simpleRNANotation)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;This method returns the nucleotide sequence for a single simple RNA notation</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B>NucleotideConverter.</B><B><A HREF="../../../../org/helm/notation/tools/NucleotideConverter.html#getNucleotideSequencesFromComplexNotation(java.lang.String)">getNucleotideSequencesFromComplexNotation</A></B>(java.lang.String&nbsp;complexNotation)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;This method returns the nucleotide sequences for nucleic acid polymer mixture</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B>ComplexNotationParser.</B><B><A HREF="../../../../org/helm/notation/tools/ComplexNotationParser.html#getOtherString(java.lang.String)">getOtherString</A></B>(java.lang.String&nbsp;extendedNotation)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B>SimpleNotationParser.</B><B><A HREF="../../../../org/helm/notation/tools/SimpleNotationParser.html#getPeptideSequence(java.lang.String)">getPeptideSequence</A></B>(java.lang.String&nbsp;polymerNotation)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;This method returns the single letter natural analog peptide sequence</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.util.List&lt;<A HREF="../../../../org/helm/notation/model/PolymerNode.html" title="class in org.helm.notation.model">PolymerNode</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>ComplexNotationParser.</B><B><A HREF="../../../../org/helm/notation/tools/ComplexNotationParser.html#getPolymerNodeList(java.lang.String)">getPolymerNodeList</A></B>(java.lang.String&nbsp;allNodeString)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;This methods converts the allNodeString to a List of PolymerNode
 Note on 2/25/2011 by Tianhong:
 CHEM nodes could contain adhoc monomers, which contains SMILES string in the format of ID?SMILES
 SMILES could have LIST-LEVEL_DELIMITER which is pipe |
 Need to find them and do a preprocess, add adhoc monomer into monomer factory</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B>NucleotideSequenceParser.</B><B><A HREF="../../../../org/helm/notation/tools/NucleotideSequenceParser.html#getReverseComplementSequence(java.lang.String)">getReverseComplementSequence</A></B>(java.lang.String&nbsp;sequence)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;returns complement sequence in reverse (3-5) direction</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B>NucleotideSequenceParser.</B><B><A HREF="../../../../org/helm/notation/tools/NucleotideSequenceParser.html#getReverseComplementSequence(java.lang.String, java.lang.String)">getReverseComplementSequence</A></B>(java.lang.String&nbsp;sequence,
                             java.lang.String&nbsp;notationSource)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B>NucleotideSequenceParser.</B><B><A HREF="../../../../org/helm/notation/tools/NucleotideSequenceParser.html#getReverseSequence(java.lang.String)">getReverseSequence</A></B>(java.lang.String&nbsp;sequence)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;This method returns the sequence in reverse direction from the input sequence, ie.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.util.List&lt;<A HREF="../../../../org/helm/notation/model/RNAPolymerNode.html" title="class in org.helm.notation.model">RNAPolymerNode</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>ComplexNotationParser.</B><B><A HREF="../../../../org/helm/notation/tools/ComplexNotationParser.html#getRNAPolymerNodeList(java.lang.String)">getRNAPolymerNodeList</A></B>(java.lang.String&nbsp;complexNotation)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B>SimpleNotationParser.</B><B><A HREF="../../../../org/helm/notation/tools/SimpleNotationParser.html#getSimpleCanonicalNotation(java.lang.String, java.lang.String)">getSimpleCanonicalNotation</A></B>(java.lang.String&nbsp;simpleNotation,
                           java.lang.String&nbsp;polymerType)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B>SimpleNotationParser.</B><B><A HREF="../../../../org/helm/notation/tools/SimpleNotationParser.html#getSimpleCanonicalNotationForChem(java.lang.String)">getSimpleCanonicalNotationForChem</A></B>(java.lang.String&nbsp;simpleNotation)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B>SimpleNotationParser.</B><B><A HREF="../../../../org/helm/notation/tools/SimpleNotationParser.html#getSimpleCanonicalNotationForPeptide(java.lang.String)">getSimpleCanonicalNotationForPeptide</A></B>(java.lang.String&nbsp;simpleNotation)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B>SimpleNotationParser.</B><B><A HREF="../../../../org/helm/notation/tools/SimpleNotationParser.html#getSimpleCanonicalNotationForRNA(java.lang.String)">getSimpleCanonicalNotationForRNA</A></B>(java.lang.String&nbsp;simpleNotation)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.util.Map.Entry&lt;java.lang.Integer,java.lang.String&gt;</CODE></FONT></TD>
<TD><CODE><B>SimpleNotationParser.</B><B><A HREF="../../../../org/helm/notation/tools/SimpleNotationParser.html#getSimpleCanonicalNotationMapEntry(java.lang.String, java.lang.String)">getSimpleCanonicalNotationMapEntry</A></B>(java.lang.String&nbsp;simpleNotation,
                                   java.lang.String&nbsp;polymerType)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B>SimpleNotationParser.</B><B><A HREF="../../../../org/helm/notation/tools/SimpleNotationParser.html#getSimplePolymerSMILES(java.lang.String, java.lang.String)">getSimplePolymerSMILES</A></B>(java.lang.String&nbsp;polymerNotation,
                       java.lang.String&nbsp;polymerType)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;This method generates the SMILES string for simple polymer notation</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;<A HREF="../../../../org/helm/notation/model/RgroupStructure.html" title="class in org.helm.notation.model">RgroupStructure</A></CODE></FONT></TD>
<TD><CODE><B>SimpleNotationParser.</B><B><A HREF="../../../../org/helm/notation/tools/SimpleNotationParser.html#getSimplePolymerStructure(java.lang.String, java.lang.String)">getSimplePolymerStructure</A></B>(java.lang.String&nbsp;polymerNotation,
                          java.lang.String&nbsp;polymerType)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;This method generates the RgroupStructure of simple polymer notation</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B>NucleotideConverter.</B><B><A HREF="../../../../org/helm/notation/tools/NucleotideConverter.html#getSimpleRNANotationFromSingleNucleotideSequence(java.lang.String)">getSimpleRNANotationFromSingleNucleotideSequence</A></B>(java.lang.String&nbsp;singleOligoSequence)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;This method returns the simple RNA notation for a single oligonucleotide sequence</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B>NucleotideSequenceParser.</B><B><A HREF="../../../../org/helm/notation/tools/NucleotideSequenceParser.html#getSirnaNotation(java.lang.String, java.lang.String)">getSirnaNotation</A></B>(java.lang.String&nbsp;senseSeq,
                 java.lang.String&nbsp;antiSenseSeq)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;This method converts nucleotide sequences into HELM Notation</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B>NucleotideSequenceParser.</B><B><A HREF="../../../../org/helm/notation/tools/NucleotideSequenceParser.html#getSirnaNotation(java.lang.String, java.lang.String, java.lang.String)">getSirnaNotation</A></B>(java.lang.String&nbsp;senseSeq,
                 java.lang.String&nbsp;antiSenseSeq,
                 java.lang.String&nbsp;rnaDesignType)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;this method converts nucleotide sequences into HELM notation based on design pattern</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.util.List&lt;<A HREF="../../../../org/helm/notation/model/Nucleotide.html" title="class in org.helm.notation.model">Nucleotide</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>SimpleNotationParser.</B><B><A HREF="../../../../org/helm/notation/tools/SimpleNotationParser.html#getStrictNucleotideList(java.lang.String, boolean)">getStrictNucleotideList</A></B>(java.lang.String&nbsp;polymerNotation,
                        boolean&nbsp;validate)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;getNucleotideList is a more forgiving function that will try to substitute
 the natural sequence where possible to provide information for queries.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;int</CODE></FONT></TD>
<TD><CODE><B>ComplexNotationParser.</B><B><A HREF="../../../../org/helm/notation/tools/ComplexNotationParser.html#getTotalMonomerCount(java.lang.String)">getTotalMonomerCount</A></B>(java.lang.String&nbsp;notation)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B>SimpleNotationParser.</B><B><A HREF="../../../../org/helm/notation/tools/SimpleNotationParser.html#getTrimmedNucleotideSequence(java.lang.String)">getTrimmedNucleotideSequence</A></B>(java.lang.String&nbsp;polymerNotation)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;generate single letter (natural analog) nucleotide sequence for simple RNA notation, remove nucleotide without base from both ends</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B>ComplexNotationParser.</B><B><A HREF="../../../../org/helm/notation/tools/ComplexNotationParser.html#hasNucleotideModification(java.util.List)">hasNucleotideModification</A></B>(java.util.List&lt;<A HREF="../../../../org/helm/notation/model/PolymerNode.html" title="class in org.helm.notation.model">PolymerNode</A>&gt;&nbsp;polymerNodes)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns true if the any of the polymerNodes that are not nucleotides have
 has a modified nucleotide.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B>ComplexNotationParser.</B><B><A HREF="../../../../org/helm/notation/tools/ComplexNotationParser.html#hybridize(java.lang.String)">hybridize</A></B>(java.lang.String&nbsp;complexNotation)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;This method will automatically add base pair info into notation only if it contains TWO RNA polymer nodes and there is no base pairing info</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B>NucleotideSequenceParser.</B><B><A HREF="../../../../org/helm/notation/tools/NucleotideSequenceParser.html#isInOppositeDirection(java.lang.String, java.lang.String)">isInOppositeDirection</A></B>(java.lang.String&nbsp;senseSeq,
                      java.lang.String&nbsp;antiSenseSeq)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;<A HREF="../../../../org/helm/notation/model/PolymerEdge.html" title="class in org.helm.notation.model">PolymerEdge</A></CODE></FONT></TD>
<TD><CODE><B>EdgeParser.</B><B><A HREF="../../../../org/helm/notation/tools/EdgeParser.html#parse(java.lang.String)">parse</A></B>(java.lang.String&nbsp;edgeNotation)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;parses input edgeNotation into a PolymerEdge object</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;<A HREF="../../../../org/helm/notation/model/ComplexPolymer.html" title="class in org.helm.notation.model">ComplexPolymer</A></CODE></FONT></TD>
<TD><CODE><B>ComplexNotationParser.</B><B><A HREF="../../../../org/helm/notation/tools/ComplexNotationParser.html#parse(java.lang.String)">parse</A></B>(java.lang.String&nbsp;extendendNotation)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;This method convert the complex polymer extendendNotation into an ComplexPolymer object
 Will throw NotationException if graph extendendNotation format is invalid,</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>protected static&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B>SimpleNotationParser.</B><B><A HREF="../../../../org/helm/notation/tools/SimpleNotationParser.html#preprocessChemNode(java.lang.String)">preprocessChemNode</A></B>(java.lang.String&nbsp;nodeDesc)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;This method preprocess the CHEMCHEM monomer node.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>protected static&nbsp;<A HREF="../../../../org/helm/notation/model/MoleculeInfo.html" title="class in org.helm.notation.model">MoleculeInfo</A></CODE></FONT></TD>
<TD><CODE><B>StructureParser.</B><B><A HREF="../../../../org/helm/notation/tools/StructureParser.html#processMoleculeInfo(java.util.List, java.util.List)">processMoleculeInfo</A></B>(java.util.List&lt;<A HREF="../../../../org/helm/notation/model/MoleculeInfo.html" title="class in org.helm.notation.model">MoleculeInfo</A>&gt;&nbsp;chunks,
                    java.util.List&lt;<A HREF="../../../../org/helm/notation/model/MoleculeInfo.html" title="class in org.helm.notation.model">MoleculeInfo</A>&gt;&nbsp;caps)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;This method should be used by SimpleNotationParser and ComplexNotationParser, not exposed to public</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B>SimpleNotationParser.</B><B><A HREF="../../../../org/helm/notation/tools/SimpleNotationParser.html#replaceMonomer(java.lang.String, java.lang.String, java.lang.String, java.lang.String)">replaceMonomer</A></B>(java.lang.String&nbsp;simpleNotation,
               java.lang.String&nbsp;polymerType,
               java.lang.String&nbsp;existingMonomerID,
               java.lang.String&nbsp;newMonomerID)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;This method replaces existing monomer with new monomer in simple polymer notation</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B>ComplexNotationParser.</B><B><A HREF="../../../../org/helm/notation/tools/ComplexNotationParser.html#replaceMonomer(java.lang.String, java.lang.String, java.lang.String, java.lang.String)">replaceMonomer</A></B>(java.lang.String&nbsp;complexNotation,
               java.lang.String&nbsp;polymerType,
               java.lang.String&nbsp;existingMonomerID,
               java.lang.String&nbsp;newMonomerID)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;This method replace existing monomer with new monomer for a given polymer type in the complex notation</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.util.List&lt;<A HREF="../../../../org/helm/notation/model/PolymerEdge.html" title="class in org.helm.notation.model">PolymerEdge</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>EdgeParser.</B><B><A HREF="../../../../org/helm/notation/tools/EdgeParser.html#sort(java.util.List, java.util.List)">sort</A></B>(java.util.List&lt;<A HREF="../../../../org/helm/notation/model/PolymerEdge.html" title="class in org.helm.notation.model">PolymerEdge</A>&gt;&nbsp;canonicalPolymerEdges,
     java.util.List&lt;java.lang.String&gt;&nbsp;sortedNodeIDs)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;sort canonical polymer edges based on sorted nodes</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B>ComplexNotationParser.</B><B><A HREF="../../../../org/helm/notation/tools/ComplexNotationParser.html#standardize(java.lang.String)">standardize</A></B>(java.lang.String&nbsp;notation)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Standardize all forms of HELM notation into complex notation</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B>ComplexNotationParser.</B><B><A HREF="../../../../org/helm/notation/tools/ComplexNotationParser.html#validateComplexNotation(java.lang.String)">validateComplexNotation</A></B>(java.lang.String&nbsp;extendedNotation)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;This methods validates the complex polymer extendendNotation</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B>ComplexNotationParser.</B><B><A HREF="../../../../org/helm/notation/tools/ComplexNotationParser.html#validateComplexPolymer(org.helm.notation.model.ComplexPolymer)">validateComplexPolymer</A></B>(<A HREF="../../../../org/helm/notation/model/ComplexPolymer.html" title="class in org.helm.notation.model">ComplexPolymer</A>&nbsp;complexPolymer)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;This methods checks Nodes and Edges matches, and checks validity of each node label</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>protected static&nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B>SimpleNotationParser.</B><B><A HREF="../../../../org/helm/notation/tools/SimpleNotationParser.html#validateMonomerReplacement(java.lang.String, java.lang.String, java.lang.String)">validateMonomerReplacement</A></B>(java.lang.String&nbsp;polymerType,
                           java.lang.String&nbsp;existingMonomerID,
                           java.lang.String&nbsp;newMonomerID)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B>ComplexNotationParser.</B><B><A HREF="../../../../org/helm/notation/tools/ComplexNotationParser.html#validateNotationFormat(java.lang.String)">validateNotationFormat</A></B>(java.lang.String&nbsp;extendedNotation)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B>SimpleNotationParser.</B><B><A HREF="../../../../org/helm/notation/tools/SimpleNotationParser.html#validateSimpleNotation(java.lang.String, java.lang.String)">validateSimpleNotation</A></B>(java.lang.String&nbsp;polymerNotation,
                       java.lang.String&nbsp;polymerType)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;This method checks if simple polymer notation is valid</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B>SimpleNotationParser.</B><B><A HREF="../../../../org/helm/notation/tools/SimpleNotationParser.html#validateSimpleNotationForChem(java.lang.String)">validateSimpleNotationForChem</A></B>(java.lang.String&nbsp;polymerNotation)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;validate chem simple notation</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B>SimpleNotationParser.</B><B><A HREF="../../../../org/helm/notation/tools/SimpleNotationParser.html#validateSimpleNotationForPeptide(java.lang.String)">validateSimpleNotationForPeptide</A></B>(java.lang.String&nbsp;polymerNotation)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;validate peptide simple notation</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B>SimpleNotationParser.</B><B><A HREF="../../../../org/helm/notation/tools/SimpleNotationParser.html#validateSimpleNotationForRNA(java.lang.String)">validateSimpleNotationForRNA</A></B>(java.lang.String&nbsp;polymerNotation)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;validate RNA simple notation</TD>
</TR>
</TABLE>
&nbsp;
<P>
<HR>


<!-- ======= START OF BOTTOM NAVBAR ====== -->
<A NAME="navbar_bottom"><!-- --></A>
<A HREF="#skip-navbar_bottom" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_bottom_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../package-summary.html"><FONT CLASS="NavBarFont1"><B>Package</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../org/helm/notation/NotationException.html" title="class in org.helm.notation"><FONT CLASS="NavBarFont1"><B>Class</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Use</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../package-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../index-files/index-1.html"><FONT CLASS="NavBarFont1"><B>Index</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
<b>HELMNotationToolkit<br><font size='-1'>1.0.1</font></b></EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;PREV&nbsp;
&nbsp;NEXT</FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../../../../index.html?org/helm/notation/\class-useNotationException.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="NotationException.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../../../../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../../../../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_bottom"></A>
<!-- ======== END OF BOTTOM NAVBAR ======= -->

<HR>
Copyright  2013, The Pistoia Alliance
</BODY>
</HTML>
